src/components/SignForm/SignUp/index.js

```diff
@@ -69,86 +50,92 @@ const SignUp = ({ setFormType }) => {
+      default: '비밀번호',
+      error: '비밀번호는 4글자 이상이어야 합니다',
+      check: () => inputState.password.length <= 3,
+      type: 'password',
+    },
+    CHECKED_PASSWORD: {
+      inputState: inputState.checkedPassword,
+      focusState: focusState.checkedPasswordFocus,
+      ref: checkedPasswordRef,
+      default: '비밀번호 확인',
+      error: '비밀번호와 일치하지 않습니다.',
+      check: () => !(inputState.password === inputState.checkedPassword),
+      type: 'password',
+    },
+    NAME: {
+      inputState: inputState.name,
+      focusState: focusState.nameFocus,
+      ref: nameRef,
+      default: '이름',
+      error: '이름은 2글자 이상이어야 합니다',
+      check: () => inputState.name.length <= 1,
+    },
+    STUDENT_ID: {
+      inputState: inputState.studentId,
+      focusState: focusState.studentIdFocus,
+      ref: studentIdRef,
+      default: '학번',
+      error: '10자리의 학번 형식이어야 합니다.',
+      check: () => inputState.studentId.length !== 10,
+    },
+    BAEKJOON_ID: {
+      inputState: inputState.baekjoonId,
+      focusState: focusState.baekjoonIdFocus,
+      ref: baekjoonIdRef,
+      default: '백준 닉네임',
+      error: '백준 닉네임은 2글자 이상이어야 합니다',
+      check: () => inputState.baekjoonId.length <= 1,
+    },
+    DISCORD_ID: {
+      inputState: inputState.discordId,
+      focusState: focusState.discordIdFocus,
+      ref: discordIdRef,
+      default: '디스코드 아이디',
+      error: '디스코드 아이디는 2글자 이상이어야 합니다',
+      check: () => inputState.discordId.length <= 1,
+    },
+    NOTION_EMAIL: {
+      inputState: inputState.notionEmail,
+      focusState: focusState.notionEmailFocus,
+      ref: notionEmailRef,
+      default: '노션 이메일',
+      error: '이메일 형식이 잘못됐습니다',
+      check: () => !isValidEmail(inputState.notionEmail),
+    },
+  };
 
-  const reset = type => {
-    setInputs(prev => {
-      return {
-        ...prev,
-        [type]: '',
-      };
-    });
-    setFocus(prev => {
-      return {
-        ...prev,
-        [type + 'Focus']: true,
-      };
+  useEffect(() => {
+    Object.entries(inputObject).map(([key, value]) => {
+      if (value.ref) {
+        value.ref.current.placeholder = inputObject[key].default;
+      }
     });
-  };
+  }, []);
 
   const onChange = e => {
     const { name, value } = e.target;
-    setInputs({
-      ...inputs,
-      [name]: value,
-    });
-  };
-
-  const checkPassword = () => {
-    return password === checkedPassword;
+    dispatchText({ type: name, text: value });
   };
 
-  function isValidEmail(email) {
-    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
-    return emailRegex.test(email);
-  }
-
   const onSubmit = async event => {
     event.preventDefault();
-    if (checkForm()) {
-      await LoginAPI.handleOnSubmitSignupForm(
-        name,
-        password,
-        githubId,
-        baekjoonId,
-        studentId,
-        discordId,
-        notionEmail
-      )
+    if (checkForm(inputObject, dispatchText, dispatchFocus)) {
+      await LoginAPI.handleOnSubmitSignupForm(inputState)
```

Comment:
```suggestion
      await LoginAPI.handleOnSubmitSignUpForm(inputState)
```
u도 대문자로 변경해야할 것 같아요!